packages/react-components/react-table/Spec.md

```diff
@@ -2,62 +2,373 @@
+
+Column resizing is only supported by the v8 DetailsList. Columns can be resized without any additional configuration.
+In order to support this, every column always has an inline style where the width in pixels is applied.
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+```tsx
+<Table>
+  <TableHeader>
+    <TableRow>
+      <TableCell> </TableCell>
+      <TableCell> </TableCell>
+      <TableCell> </TableCell>
+    </TableRow>
+  </TableHeader>
+
+  <TableBody>
+    <TableRow>
+      <TableCell> </TableCell>
+      <TableCell> </TableCell>
+      <TableCell> </TableCell>
+    </TableRow>
+  </TableBody>
+</Table>
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+### Sorting
+
+The table header should support design and interactions for sorting the table by column.
+
+### Selectable rows
+
+Table rows can be selected. The table header should also support a 'select all' feature, selecting the
+table header should select all rows.
+
+### Primary column
+
+A primary column is generally the first column of the table, however there is no strict requirement on this. This
+column has some design differences and supports secondary content which can contain extra instructions or
+description.
+
+### Sizes
+
+The table supports the following sizes that affect the layout and size of its child components:
+
+- default
+- small
+- smaller
+- large
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+Current proposed API, will be eventually be replaced with a hard link to the `Component.types.ts` file
+more mature into development.
+
+```ts
+interface TableProps {
+  /**
+   * Determines sizes of child components
+   */
+  size: 'small' | 'smaller' | 'large';
+  /**
+   * Whether the columns are sortable
+   */
+  sortable?: boolean;
+
+  /**
+   * Determines if/how the table can be navigated by keyboard
+   *
+   * @default none
+   */
+  navigationMode?: 'none' | 'cell' | 'row' | 'composite';
+}
+
+interface TableHeaderCell {
+  /**
+   * Whether the column is sortable overrides the top level table prop
+   */
+  sortable: boolean;
+  /**
+   * How the column data is sorted
+   */
+  sortDirection: 'ascending' | 'descending';
+}
+
+/**
+ * Layout and design sugar for the primary column of a table
+ */
+interface TablePrimaryCell {
+  secondaryContent: Slot<'span'>;
+  icon: Slot<'span'>;
+}
+```
 
 ## Structure
 
-- _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
+### Basic
 
-## Migration
+```tsx
+<Table>
+  <TableHeader>
+    <TableRow>
+      <TableCell> </TableCell>
+    <TableRow>
+  </TableHeader>
+
+  <TableBody>
+    <TableRow>
+      <TableCell> </TableCell>
+    </TableRow>
+  </TableBody>
+</Table>
+```
+
+```html
+<div role="table">
```

Comment:
I guess, if there's going to be a separation of a static vs interactive table then I don't mind using semantic table elements for the static table. Anything interactive will need to role `grid` and `gridcell` so that can end up as divs.